docs/source/tutorial.rst

```diff
@@ -0,0 +1,553 @@
+
+    teardown() {
+        rm -f /tmp/bats-tutorial-project-ran
+    }
+
+Now running the test again first give us the same error, as the teardown has not run yet. 
+On the second try we get a clean `/tmp` folder again and our test passes consistently now.
+
+It is worth noting that we could do this `rm` in the test code itself but it would get skipped on failures.
+
+.. important::
+
+    A test ends at its first failure. Non of the following commands in this test will be executed.
+    The `teardown` function runs after each individual test in a file, regardless of test success or failure.
+    Similarly to `setup`, each `.bats` file can have its own `teardown` function which will be the same for all tests in the file.
+
+Test what you can
+-----------------
+
+Sometimes tests rely on the environment to provide infrastructure that is needed for the test.
+If not all test environments provide this infrastructure but we still want to test on them,
+it would be unhelpful to get errors on parts that are not testable.
+
+Bats provides you with the `skip` command which can be used in `setup` and `test`.
+
+.. tip::    
+    
+    You should `skip` as early as you know it does not make sense to continue.
+
+In our example project we rewrite the welcome message test to `skip` instead of doing cleanup:
+
+.. code-block:: bash
+
+    teardown() {
+        : # Look Ma! No cleanup!
+    } 
+
+    @test "Show welcome message on first invocation" {
+        if [[ -e /tmp/bats-tutorial-project-ran ]]; then
+            skip 'The FIRST_RUN_FILE already exists'
+        fi
+        
+        run project.sh
+        assert_output --partial 'Welcome to our project!'
+
+        run project.sh
+        refute_output --partial 'Welcome to our project!'
+    }
+
+The first test run still works due to the cleanup from the last round. However, our second run gives us:
+
+.. code-block:: console
+
+    $ ./test/bats/bin/bats test/test.bats 
+     - Show welcome message on first invocation (skipped: The FIRST_RUN_FILE already exists)
+
+    1 test, 0 failures, 1 skipped
+
+.. important::
+
+    Skipped tests won't fail a test suite and are counted separately.
+    No test command after `skip` will be executed. If an error occurs before `skip`, the test will fail.
+    An optional reason can be passed to `skip` and will be printed in the test output.
+
+Setting up a multifile test suite
+---------------------------------
+
+With a growing project, putting all tests into one file becomes unwieldy.
+For our example project, we will extract functionality into the additional file `src/helper.sh`:
+
+,, code-block:: bash
+
+#!/usr/bin/env bash
+
+    _is_first_run() {
+        local FIRST_RUN_FILE=/tmp/bats-tutorial-project-ran
+        if [[ ! -e "$FIRST_RUN_FILE" ]]; then
+            touch "$FIRST_RUN_FILE"
+            return 0
+        fi
+        return 1
+    }
+
+This allows for testing it separately in a new file `test/helper.bats`:
+
+.. code-block:: bash
```

Comment:
Annoyingly this code highlighting doesn't seem to work, is it the shebang that does it? The next block is ok, for example

Reply:
There was an error in the code which probably derailed the parser.